lib/assessment/label.py

```diff
@@ -358,15 +357,22 @@ def compute_messages(self, prompt, rubric, student_code, examples=[]):
         return messages
 
     def get_response_data_if_valid(self, choice, rubric, student_id, choice_index=None, reraise=False, response_type='tsv'):
-        response_text = choice['message']['content']
+        response_text = None
+        finish_reason = None
+        if 'message' in choice.keys(): # OpenAI response
+            response_text = choice['message']['content']
+            finish_reason = choice['finish_reason']
+        elif 'completion' in choice.keys(): # Claude response
+            response_text = choice['completion']
+            finish_reason = choice['stop_reason']
         try:
             choice_text = f"Choice {choice_index}: " if choice_index is not None else ''
             if not response_text:
-                raise InvalidResponseError("empty response")
+                raise InvalidResponseError(f"empty or invalid response:\n{choice}")
             text = response_text.strip()
 
             if response_type == 'json':
-                response_data = self.parse_json_response(text, student_id, finish_reason=choice['finish_reason'])
+                response_data = self.parse_json_response(text, student_id, finish_reason=finish_reason)
```

Comment:
The finish reason is used to produce an error on the CDO side if the response was terminated early due to exceeding the token limit (prompt+response length), usually because the student code is too long.

We should either map the finish_reason/ stop_reason strings here so AWS and OpenAI models have the same set of finish reason strings, or make a corresponding PR in CDO to accept either AWS or OpenAI strings to produce the error

Reply:
wait, it looks like the error is raised in `self.parse_json_response` on lines 398-403:
```
    def parse_json_response(self, response_text, student_id, finish_reason):
        # capture all data from the first '[' to the last ']', inclusive
        match = re.search(r'(\[.*\])', response_text,re.DOTALL)
        if not match:
            if finish_reason == 'length':
                raise RequestTooLargeError(f"{student_id}: no valid JSON data")
```
IMO this is where you should also accept whatever the corresponding AWS termination reason is to `'length'`

Reply:
OK, this should be an easy fix. I'll look at what Claude outputs as finish reasons and add that mapping

Reply:
Corresponding string for Claude is `max_tokens` https://docs.anthropic.com/en/api/complete